superduper/jobs/queue.py

```diff
@@ -170,33 +171,40 @@ def consume(self, db: 'Datalayer', queue: t.Dict, components: t.Dict):
         """Consume the current queue and run jobs."""
         from superduper.base.datalayer import Event
 
-        queue_jobs: t.Dict[str, t.List] = {}
-        for component_id in queue:
-            events = queue[component_id]
+        queue_jobs = defaultdict(lambda: [])
+        components_to_use = [('listener', x) for x in db.show('listener')]
```

Comment:
We need a more general feature of "trigger".

Reply:
I optimized the logic of the trigger part, but we can merge it first with some minor modifications. I’ll make adjustments based on your PR.